File: 169-majority-element/169-majority-element.cpp

class Solution {
public:
    int majorityElement(vector<int>& nums) {

    unordered_map<int , int> maap;
    for (auto x:nums)
    {
        maap[x]++ ;
    }
    int k=0;
  int max=0;
    for (auto x : maap){

      if(x.second>k){
          k=x.second;
          max=x.first;
      }}
      return max;
}};

    

File: 169-majority-element/169-majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        temp=9999999999
        for i in nums:
            if temp == i:
                count+=1
                continue
            if count<=0:
                temp = i
                count+=1
                continue
            if count > 0:
                count-=1
        return temp